File: Posdea.Application/Models/UserSegment/UserModel.cs
using AutoMapper;
using Posdea.Application.Mappings;
using Posdea.Domain.Entities.UserSegment;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Posdea.Application.Models.UserSegment
{
    public class UserModel : IMapFrom<User>
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Email { get; set; }
        [Required]
        public string PhoneNumber { get; set; }
        [Required]
        public string Password { get; set; }
        public string Status { get; set; }
        [Required]
        public AddressModel Address { get; set; }
        public int? AddressId { get; set; }
        public RoleModel? Role { get; set; }
        public int RoleId { get; set; }
    }
}

File: Posdea.WebUI/src/app/models/user.model.ts
import { Address } from "./address.model";
import { Role } from "./role.model";

export class User {
  public id! : number;
  public name! : string;
  public email! : string;
  public avatar! : string;
  public phoneNumber! : string;
  public password! : string;
  public status! : string;
  public address! : Address;
  public addressId! : number;
  public role! : Role;
  public roleId! : number;

  constructor(){}
}